src/content/1.0.0/library/clojure/clojure-overview.md

```diff
@@ -0,0 +1,208 @@
+                             :_predicate/type :ref
+                             :_predicate/multi true
+                             :_predicate/restrictCollection :author}
+                            {:_id :_predicate
+                             :_predicate/name :book/tags
+                             :_predicate/doc "Tags for categorizing books."
+                             :_predicate/type :tag
+                             :_predicate/multi true}])
+
+```
+
+#### Data
+
+Insert some subject data that conforms to the schema we've created. See the [Transact](/docs/1.0.0/transact) documentation for more details.
+
+```all
+;; adding data
+@(fdb/transact conn ledger [{:_id :book
+                             :book/title "Watership Down"
+                             :book/tags [:rabbits :fiction]
+                             :book/authors [{:_id :author :name "Richard Adams"}]}
+                            {:_id :book
+                             :book/title "The Joy of Clojure"
+                             :book/tags [:programming :non-fiction]
+                             :book/authors [{:_id :author :name "Michael Fogus"}
+                                       {:_id :author :name "Chris Houser"}]}
+                            {:_id :book
+                             :book/title "Cryptonomicon"
+                             :book/tags [:fiction :sci-foo]
+                             :book/authors [{:_id :author :name "Neal Stephenson"}]}])
+
+@(fdb/transact conn ledger [{:_id [:book/title "Cryptonomicon"]
+                             :book/notes "I rly liked it."}])
+
+;; updating a single-cardinality predicate
+@(fdb/transact conn ledger [{:_id [:book/title "Cryptonomicon"]
+                             :book/notes "I really liked it."}])
+
+;; updating a multi-cardinality predicate
+@(fdb/transact conn ledger [{:_id [:book/title "The Joy of Clojure"]
+                             :book/title "The Joy of Clojure"
+                             :book/tags [:favorite]}])  ; adding the :favorite tag
+
+;; retracting a predicate
+@(fdb/transact conn ledger [{:_id [:book/title "Cryptonomicon"] :book/notes nil}])
+
+;; retracting multi-cardinality member
+@(fdb/transact conn ledger [{:_id [:book/title "The Joy of Clojure"]
+                             :_action :delete
+                             :book/tags [:favorite]}]) ; only removing :favorite tag
+
+;; retracting a whole subject
+@(fdb/transact conn ledger [{:_id [:book/title "Watership Down"] :_action :delete}])
+
+```
+
+### Querying
+
+#### Analytical queries
+
+The syntax for these queries is similar to datalog and eql. These don't support keywords for matching on flake values in the `:where` clause.
+
+The `query` function takes a `db` value to execute a query against, as well as a query map. See the docs for [Analytical queries](/docs/1.0.0/query/analytical-query) and [Basic queries](/docs/1.0.0/query/overview) for more details.
+
+```all
+;; all books
```

Comment:
I'd suggest perhaps adding above this:

;; store the current immutable database as a variable. See API docs for getting a prior version of a db, applying permissions to a db, and other options.

(def db (fdb/db conn ledger))

Reply:
I got a little bitten by this as I was first learning the api - I had `def`ed the db value and I was bouncing back between transacting and querying and never seeing any of my updates until I realized that the `(db ,,,)` call gives you an immutable snapshot. I like adding this verbiage because if I'd read about it then I wouldn't have been surprised, but I think we should still use the full `(fdb/db conn ledger)` in the example queries instead of hardcoding the cached db value.